resin/settings.py

```diff
@@ -35,15 +36,20 @@ class Settings(object):
                                             'cache')
 
     def __init__(self):
-        if Path.isdir(self._setting['data_directory']):
+        try:
+            self.__read_settings()
+        except:
+            # move old settings file if it exists.
             if Path.isfile(Path.join(self._setting['data_directory'],
                                      self.CONFIG_FILENAME)):
-                self.__read_settings()
-            else:
-                self.__write_settings()
-        else:
-            # data directory doesn't exist
+                shutil.move(Path.join(self._setting['data_directory'],
+                                     self.CONFIG_FILENAME)
+                            , Path.join(self._setting['data_directory'],
+                                     "{0}.{1}".format(self.CONFIG_FILENAME,'old')))
             self.__write_settings()
+            raise exceptions.InvalidSettings(Path.join(
+                                            self._setting['data_directory'],
+                                            self.CONFIG_FILENAME))
```

Comment:
I don't see how raising an exception is helpful here. This code will crash once (when it finds sth is wrong with the settings), but will then happily start and keep running if the user runs a command again (because you fixed the problem by writing a proper settings file). You're backing up the previous file, so I think it's more appropriate to issue a warning and keep going.